rust/extractor/src/translate/base.rs

```diff
@@ -170,20 +169,37 @@ impl<'a> Translator<'a> {
         location: (LineCol, LineCol),
     ) {
         let (start, end) = location;
-        let level = match severity {
-            DiagnosticSeverity::Debug => Level::Debug,
-            DiagnosticSeverity::Info => Level::Info,
-            DiagnosticSeverity::Warning => Level::Warn,
-            DiagnosticSeverity::Error => Level::Error,
+        match severity {
+            DiagnosticSeverity::Debug => tracing::debug!(
+                "{}:{}:{}: {}",
+                self.path,
+                start.line + 1,
+                start.col + 1,
+                &full_message
+            ),
+            DiagnosticSeverity::Info => tracing::info!(
+                "{}:{}:{}: {}",
+                self.path,
+                start.line + 1,
+                start.col + 1,
+                &full_message
+            ),
+            DiagnosticSeverity::Warning => tracing::warn!(
+                "{}:{}:{}: {}",
+                self.path,
+                start.line + 1,
+                start.col + 1,
+                &full_message
+            ),
+            DiagnosticSeverity::Error => tracing::error!(
+                "{}:{}:{}: {}",
+                self.path,
+                start.line + 1,
+                start.col + 1,
+                &full_message
+            ),
```

Comment:
you can avoid all this duplication by using `tracing::Level` and replacing the old `log:log!` with `tracing:event!`

Reply:
I tried, but that does not seem to be possible. See also: https://github.com/tokio-rs/tracing/issues/2730

Reply:
ah, I see. Too bad :/

Reply:
This might call for a local macro definition along the lines of https://github.com/tokio-rs/tracing/issues/2730#issuecomment-1943022805:
```suggestion
        // see https://github.com/tokio-rs/tracing/issues/2730
        macro_rules! dispatch_to_tracing {
            ($lvl:ident, $($arg:tt)+) => {
                match $lvl {
                    DiagnosticSeverity::Debug => ::tracing::debug!($($arg)+),
                    DiagnosticSeverity::Info => ::tracing::info!($($arg)+),
                    DiagnosticSeverity::Warning => ::tracing::warn!($($arg)+),
                    DiagnosticSeverity::Error => ::tracing::error!($($arg)+),
                }
            };
        }

        dispatch_to_tracing!(
                severity,
                "{}:{}:{}: {}",
                self.path,
                start.line + 1,
                start.col + 1,
                &full_message
        );
```